app/src/main/java/de/westnordost/streetcomplete/quests/baby_changing_table/AddBabyChangingTable.kt

```diff
@@ -13,11 +13,9 @@ class AddBabyChangingTable : OsmFilterQuestType<Boolean>() {
     override val elementFilter = """
         nodes, ways with
         (
-          (
-            (amenity ~ restaurant|cafe|fuel|fast_food or shop ~ mall|department_store)
-            and toilets = yes
-          )
-          or amenity = toilets
+          amenity ~ restaurant|cafe|toilets
+          or amenity = fuel or shop ~ mall|department_store and toilets = yes
```

Comment:
```suggestion
          or (amenity = fuel or shop ~ mall|department_store) and toilets = yes
```

If the intention is to ask only on fuel stations that have toilets, then there are missing parenthesis (as suggested here).

If the intention is instead to ask on all fuel stations regardless of presence of toilets, then it should move to that regexp above (with restaurants) to be more readable.

Reply:
Thanks for pointing that our! It was not my intention to ask for all fuel stations, therefore i applied you code proposal.

My main intention is about restaurants, cafés and fast_food.

I am rebuilting a dev version right now and will test this again.